tests/php/test-class-amp-theme-support.php

```diff
@@ -1424,6 +1424,49 @@ public function test_scripts_get_moved_to_head() {
 		}
 	}
 
+	/**
+	 * Test removing duplicate scripts.
+	 *
+	 * @covers AMP_Theme_Support::prepare_response()
+	 * @covers AMP_Theme_Support::ensure_required_markup()
+	 */
+	public function test_duplicate_scripts_are_removed() {
+		wp();
+		add_theme_support( AMP_Theme_Support::SLUG );
+		AMP_Theme_Support::init();
+		AMP_Theme_Support::finish_init();
+
+		ob_start();
+		?>
+		<html>
+			<head>
+				<?php wp_print_scripts( [ 'amp-runtime', 'amp-mustache', 'amp-list', 'amp-runtime', 'amp-mustache', 'amp-list' ] ); ?>
```

Comment:
WordPress's dependencies system automatically skip printing dependencies that have already been printed, so duplicating handles won't have any effect here. Nor will printing the same thing in the footer have any effect.